rust/kernel/sync/arc.rs

```diff
@@ -49,8 +54,19 @@ unsafe impl<T: RefCounted + ?Sized + Sync + Send> Sync for Ref<T> {}
 impl<T: RefCounted> Ref<T> {
     /// Constructs a new reference counted instance of `T`.
     pub fn try_new(contents: T) -> Result<Self> {
-        let boxed = Box::try_new(contents)?;
-        boxed.get_count().count.store(1, Ordering::Relaxed);
+        Self::try_new_and_init(contents, |_| {})
```

Comment:
This feels odd. I vaguely feel that the callback should really be an `Option<Fn()>`, but I don't have strong feelings about it. If you are ok with this, I am as well.

Reply:
The intent here was just to provide a cleaner `try_new` and `try_new_and_init`. So

```rust
  Ref::try_new(T::new());
  Ref::try_new_and_init(T::new(), |x| ...);
```

instead of

```rust
  Ref::try_new(T::new(), None);
  Ref::try_new(T::new(), Some(|x| ...));
```

The second option doesn't seem so bad when I write it out though... I think I'll merge them as you suggest.

Reply:
I think the current approach is okay, it is less noisy than a `Option`. I also don't think internal private function accepting `Option` is necessary, as it is pretty clear to me that `|_| {}` means a no-op.

Reply:
Ok, I'll leave it as is for now. We can always revisit later.